test/ConfigProviderTest.php

```diff
@@ -31,5 +33,47 @@ public function testReturnedArrayContainsDependencies(array $config)
     {
         $this->assertArrayHasKey('dependencies', $config);
         $this->assertInternalType('array', $config['dependencies']);
+        $this->assertArrayHasKey('factories', $config['dependencies']);
+        $this->assertInternalType('array', $config['dependencies']['factories']);
+    }
+
+
+    public function testServicesDefinedInConfigProvider()
+    {
+        $config = ($this->provider)();
+
+        $json = json_decode(
+            file_get_contents(__DIR__ . '/../composer.lock'),
+            true
+        );
+        foreach ($json['packages'] as $package) {
+            if (isset($package['extra']['zf']['config-provider'])) {
+                $configProvider = new $package['extra']['zf']['config-provider']();
+                $config = array_merge_recursive($config, $configProvider());
+            }
+        }
+
+        $config['dependencies']['services']['config'] = [
+            'authorization' => ['roles' => [], 'permissions' => []],
+        ];
```

Comment:
An empty `config` array is not fine, I tested… I don't think there should be a default value for this if that's what you mean by " it should be also defined in ConfigProvider"

Reply:
Hm... yeh, checked factory and you have right, probably we shouldn't have empty placeholders in `ConfigProvider` because we throw exceptions to notify user where the component is not configured properly.